### PR TITLE
[TR_146] Add SVG Icon System

### PR DESCRIPTION
**Pull request by jacksonrya**
---

#### Please check if the PR fulfills these requirements

- [x] The changes don't come from your master branch.

---

#### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Update the icon component to be color and size dynamic while using SVG icons provided by the UI team.

#### What is the current behavior? (You can also link to an open issue here)
Icons are appropriated from a typeface.

#### What is the new behavior? (if this is a feature change)
- Usage of custom icons that can scale and be styled with more precision.
- **Dynamically** set colors of icons for **mobile**.
- **Statically** render icons in **browser**.

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
Maybe
- Yes: Previous usage of the icon component still works.
- No: Usage of the SvgImage will break as this component was removed.
  - The package (react-native-remote-svg) used within the component was resource intensive.
  - This component might need to be recreated down the road in order to handle the differences between importing SVGs for mobile and browser.

#### Other information
- Any new SVGs will have to follow some standards in order to work with this new icon system.
  - All paths must be converted to closed paths (e.g. use Stroke to Path in vector editing software).
    - This ensures a single attribute, 'fill', determines the color of the icon without affecting icon style ratios if the icon is scaled.
  - The system currently only handles one dynamic color per icon. If an icon has more than one color (e.g. NavBar icons with red notification dots), the non-dynamic elements must have the fill attribute set to their static color. Any dynamic parts of the icon must not have a fill attribute. In order for static rendering to work, a fill color must be provided to the SVG root element (this attribute is overridden via JS).
- In order to make SVGs dynamic and lightweight for mobile, the community-respected package (react-native-svg-transformer) was installed. This requires the adding of support files, like a Metro bundling configuration and a type declaration file.

---

#### TODO
- [x] Update icon styling
- [x] Add component props
  - size
- [x] Remove import of elegant icons
- [x] Remove added dependency left over from prototype code
- [x] Consider removing SvgImage component
  - react-native-remote-svg creates a WebView for each, individual SVG. My understanding is that WebViews use a lot of resources. The alternative is react-native-svg-transformer, much more lightweight on client-side processing, 
- [x] Test 'lock' and 'unlock' deprecated icons
- [x] Test ALL icons and post image of them
- [x] Make icons dynamic colors? Can all of them use either `stroke` or `fill` color properties?
- [x] Update README icon section
  - [x] Remove old section
  - [x] Write new section describing requirements for any new SVG files, including non icons.
- [x] Research SVG rendering issues with paths tangential to viewbox (e.g. red dot cut off in menu svg at certain zoom levels)
- [x] Render all icons properly on all devices
- [x] Review exact offset distances for non-square icons (bell, menu, etc.)
- [ ] Rebase onto upstream master with new `PasswordTextInput`

---
### Usage
Any valid color can be used. Any number size can be provided. The name is an `IconName` which are mapped to icon SVGs. 
```tsx
<Icon name="qrCode" color="pink" size={24} />
```
```tsx
import { iconImports } from '@elements/Icon';
...
{
   Object.keys(iconImports).map((name: iconName) => (
      <Icon name={name} size={24} color="pink" />
   ))
}
```

---

---
> icon-preview
_Static in browser. Note: if an icon looks uncentered, it is the SVG itself. Not the component._

> dynamic-icon-preview
_Color dynamic on MOBILE_

> icon-preview_toggle
_How icon pairs replace seamlessly_

> icon-preview_navbar
_slightly cropped red dot is a browser rendering issue to be researched_

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +38,7 @@
     "react-native-reanimated": "~1.4.0",
     "react-native-remote-svg": "^2.0.6",
     "react-native-svg": "9.13.3",
+    "react-native-svg-transformer": "^0.14.3",
```
Comment (jacksonrya) [inline package.json]: to be uninstalled

**Comment by jacksonrya** [comment]
Rebase 04/07/20 18:36

File: package.json [hunk 2]
```diff
@@ -22,7 +22,10 @@
     "expo-barcode-scanner": "~8.0.0",
     "expo-constants": "~8.0.0",
     "expo-font": "~8.0.0",
+    "expo-image-picker": "~8.0.1",
```
Comment (jacksonrya) [inline package.json:25]: These were just moved to be alphabetical

File: src/screens/RegistrationScreen/DonorRegistrationScreen.tsx [hunk 3]
```diff
@@ -125,7 +125,7 @@ export default () => {
 							<TouchableWithoutFeedback
 								onPress={pickImage}
 							>
-								<Icon name="image" style={styles.icon} />
+								<Icon name="image" size={24} />
```
Comment (jacksonrya) [inline src/screens/RegistrationScreen/DonorRegistrationScreen.tsx:128]: Should standard icon sizes (18, 24) be added as exported constants from the Icon folder?

**Comment by duskcloudxu** [comment]
Started reviewing this PR, and found that it could run on web but not on ios simulator. Could you help to verify if that is a common situation?

**Comment by jacksonrya** [comment]
It's been working when tunneled to my iPhone. I'll double check

**Pull request merged**